src/client/features/enrichment/token-input.js

```diff
@@ -0,0 +1,75 @@
+//const React = require('react');
+//const h = require('react-hyperscript');
+const _ = require('lodash');
+const { ServerAPI } = require('../../services/');
+
+
+class TokenInput {
+
+  constructor(state) {
+    this.state = state;
+  }
+
+  //called onClick 'submit'
+  //seperate div text input into an array, remove all elements that have already been validated
+  //this method will be altered later depending on the type/format of the final input box (slate or some other library)
+  parseTokenList() {
+    this.state.query = document.getElementById('gene-input-box').innerText;
+    let tokenList = this.state.query.split(/\n/g);
+    //parse tokenList to remove all elements that have already been processed and logged to map 'tokenData'
+    this.state.tokenData.forEach( (value, key) => {
+      if (tokenList.includes(key)) tokenList = _.pull(tokenList, key);
+    });
+    //send only new tokens to validation
+    this.retrieveValidationAPIResult(tokenList);
+  }
+
+  //call validation service API to retrieve unrecognized tokens as an array
+  retrieveValidationAPIResult(tokensToValidate){
+    ServerAPI.enrichmentAPI({genes: _.pull(tokensToValidate,"")}, "validation").then((result) => {
```

Comment:
Promises are not returned and so can't be chained

Reply:
I think the ServerAPI.enrichmentAPI calls upon the validation index.js which then returns a promise